io/transmat/gs/gs_token.go

```diff
@@ -0,0 +1,76 @@
+package gs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/storage/v1"
+)
+
+const EnvGsAccessToken = "GS_ACCESS_TOKEN"
+const EnvGsAccountFile = "GS_SERVICE_ACCOUNT_FILE"
+
+func GetAccessToken() (*oauth2.Token, error) {
+	var err error
+	var secrets []byte
+	token := getAccessTokenFromEnv()
+	if token == nil {
+		secrets, err = getServiceAccountSecretsFromEnv()
+		if err == nil {
+			token, err = createAccessTokenFromSecrets(secrets, []string{storage.DevstorageReadWriteScope})
+		}
+	}
+	return token, err
```

Comment:
[Effective Go](https://golang.org/doc/effective_go.html#if) suggests rephrasing `if` statements to return as early as possible to reduce nesting.  I think that would make this function easier to read.